File: base/src/main/java/com/appoets/basemodule/di/component/BaseComponent.java
package com.appoets.basemodule.di.component;

import com.appoets.basemodule.base.BaseRespositary;
import com.appoets.basemodule.di.modules.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {NetworkModule.class})
public interface BaseComponent {
    void inject(BaseRespositary respositary);
}

File: base/src/main/java/com/gox/base/di/BaseComponent.kt
package com.gox.base.di

import com.gox.base.repository.BaseRepository
import com.gox.base.utils.RunTimePermission
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [WebServiceModule::class, AppContainerModule::class])
interface BaseComponent {
    fun inject(baseRepository: BaseRepository)

    fun inject(runtimePermission: RunTimePermission)
}